### PR TITLE
BLE, TCP and UDP Support

### PR DESCRIPTION
**Pull request by Scavanger**
Adds support for wireless connections.

Bluetooth Low Energy:
Hardware:
- CC2541 based modules like HM1X/HC08/09
- Nordic Semiconductor NRF (Adafruit BLE Shield)
- SpeedyBee adapters

For CC2541 based modules a small hack in iNav is necessary: https://github.com/iNavFlight/inav/pull/7931
See also: https://github.com/iNavFlight/inav/pull/7783

TCP and UDP
See discussion here: #1436

I have written a firmware for ESP8266 modules that allows easy configuration via a web interface.
It checks MSP frames for errors and wraps an MSP frame in one UDP/TCP packet. 
UDP runs very smoothly, TCP is a bit laggy.
https://github.com/Scavanger/MSPWifiBridge
But any device that forwards the data of a UART one-to-one to TCP/UDP should work the same way. 

if #1447/https://github.com/iNavFlight/inav/pull/7812 is merged, this PR/https://github.com/iNavFlight/inav/pull/7931 may needs a rebase.

### CODE REVIEW

File: js/msp/MSPHelper.js [hunk 1]
```diff
@@ -463,22 +463,33 @@ var mspHelper = (function (gui) {
                 console.log("Servo mix saved");
                 break;
             case MSPCodes.MSP2_INAV_LOGIC_CONDITIONS:
-                LOGIC_CONDITIONS.flush();
```
Comment (DzikuVx) [inline js/msp/MSPHelper.js:466]: @Scavanger this is a great PR. But we have to change how it's done. I've left a review in the INAV repo but it ultimately goes down to:

1. Leave `MSP2_INAV_LOGIC_CONDITIONS` unchanged
1. Add `MSP2_INAV_LOGIC_CONDITIONS_SINGLE` frame
1. Switch configurator to use `MSP2_INAV_LOGIC_CONDITIONS_SINGLE`

And then, in one of future releases we can remove support for `MSP2_INAV_LOGIC_CONDITIONS` completely.

What do you think?

**Comment by Scavanger** [comment]
@DzikuVx 

Yeah, this is the cleaner way.
Should the version check for firmwares < 5.0 remain in and then be switched back to MSP2_INAV_LOGIC_CONDITIONS if necessary?

**Comment by DzikuVx** [comment]
@Scavanger yes, we'd like for Configurator 5 to support INAV 4. So it should indeed check for supported message.
If this will be to time consuming or just too complex we can always say: sorry, we do not support older version. Your call

**Review by DzikuVx — APPROVED** [review]

**Comment by DzikuVx** [comment]
@Scavanger could you please add a small doc here or in the firmware repo on how to use it? It would be awesome to have at least a short explanation of what's going on

**Comment by Scavanger** [comment]
@DzikuVx 
https://github.com/iNavFlight/inav/pull/7931

**Comment by EduardTheThird** [comment]
Tried this feature using this board, https://www.adafruit.com/product/2821,
running this code:
https://github.com/EduardTheThird/inav-configurator-ESP8266-WiFi-UART-transparent-bridge/blob/master/WiFiUartTransparentBridge/WiFiUartTransparentBridge.ino

Worked, thanks! I only had to change my baud rate to 57600.

**Pull request merged**